CMakeLists.txt

```diff
@@ -0,0 +1,256 @@
+cmake_minimum_required(VERSION 3.15.0)
+project(mapnik 
+    VERSION 3.0.24 
+    HOMEPAGE_URL "https://mapnik.org/" 
+    DESCRIPTION "Mapnik is an open source toolkit for developing mapping applications" 
+    LANGUAGES CXX 
+)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+option(BUILD_SHARED_LIBS "build mapnik dynamic" ON)
+option(BUILD_TEST "builds the tests" ON)
+option(USE_EXTERNAL_MAPBOX_GEOMETRY "Use a external mapnik/geometry.hpp. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_POLYLABEL "Use a external mapnik/polylabel. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_PROTOZERO "Use a external mapnik/protozero. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_VARIANT "Use a external mapnik/variant. If off, use the submodule" OFF)
+option(USE_JPEG "adds jpeg support" ON)
+option(USE_PNG "adds png support" ON)
+option(USE_TIFF "adds tiff support" ON)
+option(USE_WEBP "adds webp support" ON)
+option(USE_LIBXML2 "adds libxml2 support" ON)
+option(USE_CAIRO "adds the cairo renderer" ON)
+option(USE_PROJ4 "adds proj4 support" ON)
```

Comment:
`proj4` is a legacy name we should use `proj` (current version v7.2.1)  @mathisloge

Reply:
Yeah, will change to  proj 👍